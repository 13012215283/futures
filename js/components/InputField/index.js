import React, { Component } from 'react';
import { StyleSheet, TextInput } from 'react-native';
import InputFieldStyle from './inputField.style';

// @flow

const InputFieldStyles = StyleSheet.create(InputFieldStyle);
const noop = () => {};
type Props = {
  type: string,
  inputStyle?: Object,
  onChange: Function,
  onBlur: Function,
  keyboardType: string,
};

type State = {
  text: string,
};
export default class InputField extends Component<Props, State> {
  static defaultProps = {
    type: 'text',
    inputStyle: {},
    onChange: noop,
    onBlur: noop,
    keyboardType: 'numeric',
  };

  onChange(text) {
    const { onChange } = this.props;

    if (onChange) {
      onChange(text);
    }
  }

  render() {
    const { inputStyle } = this.props;
    return (
      <TextInput
        {...this.props}
        underlineColorAndroid="transparent"
        onChange={e => {
          this.onChange(e.nativeEvent.text);
        }}
        style={[InputFieldStyles.input, inputStyle]}
        placeholder="请输入"
      />
    );
  }
}
