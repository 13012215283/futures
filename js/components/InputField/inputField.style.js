import { StyleSheet } from 'react-native';
import colors from '../themes/colors';
import { px2dp } from '../../utils/px2dp';

export default {
  input: {
    width: px2dp(250),
    height: px2dp(60),
    borderWidth: StyleSheet.hairlineWidth,
    padding: 0,
    borderColor: colors['1103'],
    fontSize: px2dp(28),
    color: colors['1101'],
    textAlign: 'center',
  },
};
