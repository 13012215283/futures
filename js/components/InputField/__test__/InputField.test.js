import React from 'react';
import TestRenderer from 'react-test-renderer';
import InputField from '../index';

test('render correctly', () => {
  const tree = TestRenderer.create(<InputField />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly', () => {
  const tree = TestRenderer.create(<InputField type="integer" />).toJSON();
  expect(tree).toMatchSnapshot();
});
