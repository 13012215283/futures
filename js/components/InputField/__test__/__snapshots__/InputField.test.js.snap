// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<TextInput
  allowFontScaling={true}
  inputStyle={Object {}}
  keyboardType="numeric"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="请输入"
  style={
    Array [
      Object {
        "borderColor": "#bdbdbd",
        "borderWidth": 0.5,
        "color": "#212121",
        "fontSize": 28,
        "height": 60,
        "padding": 0,
        "textAlign": "center",
        "width": 250,
      },
      Object {},
    ]
  }
  type="text"
  underlineColorAndroid="transparent"
/>
`;

exports[`render correctly 2`] = `
<TextInput
  allowFontScaling={true}
  inputStyle={Object {}}
  keyboardType="numeric"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="请输入"
  style={
    Array [
      Object {
        "borderColor": "#bdbdbd",
        "borderWidth": 0.5,
        "color": "#212121",
        "fontSize": 28,
        "height": 60,
        "padding": 0,
        "textAlign": "center",
        "width": 250,
      },
      Object {},
    ]
  }
  type="integer"
  underlineColorAndroid="transparent"
/>
`;
